RecoVertex/PrimaryVertexProducer/src/VertexTimeAlgorithmLegacy4D.cc

```diff
@@ -0,0 +1,58 @@
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "RecoVertex/PrimaryVertexProducer/interface/VertexTimeAlgorithmLegacy4D.h"
+
+#ifdef PVTX_DEBUG
+#define LOG edm::LogPrint("VertexTimeAlgorithmLegacy4D")
+#else
+#define LOG LogDebug("VertexTimeAlgorithmLegacy4D")
+#endif
+
+VertexTimeAlgorithmLegacy4D::VertexTimeAlgorithmLegacy4D(edm::ParameterSet const& iConfig, edm::ConsumesCollector& iCC)
+    : VertexTimeAlgorithmBase(iConfig, iCC) {}
+
+void VertexTimeAlgorithmLegacy4D::fillPSetDescription(edm::ParameterSetDescription& iDesc) {
+  VertexTimeAlgorithmBase::fillPSetDescription(iDesc);
+}
+
+void VertexTimeAlgorithmLegacy4D::setEvent(edm::Event& iEvent, edm::EventSetup const&){};
+
+bool VertexTimeAlgorithmLegacy4D::vertexTime(float& vtxTime, float& vtxTimeError, const TransientVertex& vtx) const {
+  const auto num_track = vtx.originalTracks().size();
+  if (num_track == 0) {
+    return false;
+  }
+
+  double sumwt = 0.;
+  double sumwt2 = 0.;
+  double sumw = 0.;
+  double vartime = 0.;
+
+  for (const auto& trk : vtx.originalTracks()) {
+    const double time = trk.timeExt();
+    const double err = trk.dtErrorExt();
+    if ((time == 0) && (err > 0.349))
```

Comment:
this value is used to select only tracks with meaningful time errors for the vertex time computation and it comes from the value used in the [TransientTrackBuilder.cc](https://github.com/cms-sw/cmssw/blob/master/TrackingTools/TransientTrack/src/TransientTrackBuilder.cc#L17) where a similar criterion is applied

Reply:
thanks for the explanation.  I propose you declare it in the same way with a `constexpr`.  A comment might also be useful.

Reply:
IMHO it would be better to have the same constant in both pieces of code, defined in a single place. What about moving the ```TransientTrackBuilder``` constant from a unnamed namespace to the builder header, and use that for the vertex code as well? In this way we are sure 1) to use the same constant consistently and 2) we will recall where it comes from. The package is anyway dependent on that ```TrackingTools``` package.